enhancements/csvless-bundles.md

```diff
@@ -24,209 +24,488 @@ status: implementable
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kiali-operator
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: kiali-operator
+  template:
+    metadata:
+      name: kiali-operator
+      labels:
+        app: kiali-operator
+        version: v1.9.1
+    spec:
+      serviceAccountName: kiali-operator
+      containers:
+      - name: ansible
+        command:
+        - /usr/local/bin/ao-logs
+        - /tmp/ansible-operator/runner
+        - stdout
+        image: quay.io/kiali/kiali-operator:v1.9.1
+        imagePullPolicy: "IfNotPresent"
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+          readOnly: true
+      - name: operator
+        image: quay.io/kiali/kiali-operator:v1.9.1
+        imagePullPolicy: "IfNotPresent"
+        volumeMounts:
+        - mountPath: /tmp/ansible-operator/runner
+          name: runner
+        env:
+        - name: WATCH_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.annotations['olm.targetNamespaces']
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
+        - name: POD_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: OPERATOR_NAME
+          value: "kiali-operator"
+      volumes:
+      - name: runner
+        emptyDir: {}
+
+```
+
+Any `CustomResourceDefinitions` found in the streams are included in the bundle as provided APIs (a.k.a. "owned CRDs").
+
+_Ex: kiali/crds.yaml_
+
+```yaml
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: kialis.kiali.io
+  labels:
+    app: kiali-operator
+spec:
+  group: kiali.io
+  names:
+    kind: Kiali
+    listKind: KialiList
+    plural: kialis
+    singular: kiali
+  scope: Namespaced
+  subresources:
+    status: {}
+  version: v1alpha1
+  versions:
+  - name: v1alpha1
+    served: true
+    storage: true
+---
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: monitoringdashboards.monitoring.kiali.io
+  labels:
+    app: kiali
+spec:
+  group: monitoring.kiali.io
+  names:
+    kind: MonitoringDashboard
+    listKind: MonitoringDashboardList
+    plural: monitoringdashboards
+    singular: monitoringdashboard
+  scope: Namespaced
+  version: v1alpha1
 
-#### Non-Inferrable
+```
 
-The following are parts of a CSV that are NOT directly inferrable from plain kubernetes resources:
+The `bundle.yaml` file consolidates all of the additional info needed by OLM to manage the life cycle of the operator. The set of additional info is exactly (w/ required info in **bold**):
 
-- Name
-- Descriptors
-- Description
-- Display Name
-- Keywords
-- Labels
-- Links
-- Maintainers
-- Maturity
-- Provider
-- Selector
-- Icon
-- Runtime Metadata
-- MinKubeVersion
-- (Future) Optional APIs
-- Replaces
-- Version
-- InstallModes
+- **name**
```

Comment:
+1

Reply:
So the implication would be that a bundle cannot be applied to cluster without an associated index. Are we cool with that?

Reply:
That's a good point. And like I said, we don't need to solve it here/now.

It might be worth thinking about having:

- a manifest bundle
- a metadata bundle
- a manifestlist that can tie those two things together?
- an index that ties them together before handing them to OLM?
- a tool that can tie them together if deploying directly to a cluster with olm, and solely use the manifest bundle if deploying to a cluster without olm?

Reply:
> What would be the benefit of not having to have one file with a name and version vs. specifying it only at catalog addition time?

One of the main benefits I can see is for the class of metadata that defines the update and dependency graphs. For example, if I want to correct certain issues with an update edge in an index, today I may need to build -- and add -- an entirely new bundle image.

See [my notes on this](https://docs.google.com/document/d/1hO5xqNcwsCtV0iYyBacLPwfpZx2P_V77RwaxWHvgdrc/edit?usp=sharing) for more thoughts.